shared/js/background/https.es6.js

```diff
@@ -95,6 +95,67 @@ class HTTPS {
         // If it falls to here, default to reqUrl
         return reqUrl
     }
+
+    /**
+     * Modify response headers on https pages to fix mixed content.
+     * 1. For main frame requests, add uprade-insecure-requests directive
+     * 2. For subrequests, edit Access-Control-Allow-Origin header if it exists
+     *    to allow resources to be loaded on the https version of site.
+     */
+    setUpgradeInsecureRequestHeader (request) {
+        // skip requests to background tabs
+        if (request.tabId === -1) return {}
+
+        // Don't alter headers if site is whitelisted or http
+        if (request.url && request.url.indexOf('https://') === 0) {
+            let headersChanged = false
+            if (request.type === 'main_frame') {
```

Comment:
We only need to add the 'upgrade-insecure-requests' CSP to the main document, as it propagates to all subrequests on the page.